wp-content/plugins/mrb-jobs-feed/core.php

```diff
@@ -96,16 +95,119 @@ function single_job_rewrite() {
      * https://codex.wordpress.org/Plugin_API/Action_Reference/admin_post_(action)
      */
     function handle_application_upload() {
```

Comment:
This methods feels like it has grown extremely large? How we could we look to improve the code organisation here? What routes are available? What are the downsides of such large methods?